### PR TITLE
Debug IOG fork on EL distro PR

### PR DESCRIPTION
**Pull request by TrevorBenson**
Test commit to kick off premerge testing in PR

### CODE REVIEW

**Comment by TrevorBenson** [comment]
Fixes the prereqs.sh success on AL2.

File: scripts/cnode-helper-scripts/cabal-build-all.sh [hunk 1]
```diff
@@ -34,6 +34,14 @@ if [[ "${PWD##*/}" == "cardano-node" ]] || [[ "${PWD##*/}" == "cardano-db-sync"
...
   cabal install bech32 cardano-addresses-cli  --overwrite-policy=always 2>&1 | tee /tmp/build-b32-caddr.log
+  rm -rf cabal.project.local
+  if [[ "$1" == "-l" ]] ; then
+    cat <<-EOF > cabal.project.local
+		${USE_SYSTEM_LIBSODIUM}
+
+		EOF
+    chmod 640 cabal.project.local
+  fi
 fi
```
Comment (scality-trevorbenson) [inline scripts/cnode-helper-scripts/cabal-build-all.sh:45]: @rdlrt Is removing the `cabal.project.local` after installing bech32 and cardano-addresses-cli viable? Will `cabal build all` still build the exact tag installed without the entries showing where they were installed from?

Comment (rdlrt) [reply scripts/cnode-helper-scripts/cabal-build-all.sh:45]: Yip, absolutely

**Comment by TrevorBenson** [comment]
Not enthused with the amount of changes required in `cabal-build-all.sh`. The steps however are directly from `cardano-node/docs/getting-started/install.md` and all 4 OS's now successfully build with system libsodium or the IOG libsodium fork.

@rdlrt Ready for review.

File: scripts/cnode-helper-scripts/cabal-build-all.sh [hunk 2]
```diff
@@ -5,46 +5,84 @@
...
+
+  export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig
+  source ~/.bashrc
+  echo "Running cabal update to ensure you're on latest dependencies.."
+  cabal update 2>&1 | tee /tmp/cabal-build.log
+  ghcup install ghc 8.10.7
+  ghcup install cabal 3.4.0.0
+  ghcup set ghc 8.10.7
+  ghcup set cabal 3.4.0.0
+  cabal configure --with-compiler=ghc-8.10.7
```
Comment (TrevorBenson) [inline scripts/cnode-helper-scripts/cabal-build-all.sh:55]:
```suggestion
  cabal configure --with-compiler=ghc-8.10.7 2>&1 | tee /tmp/cabal-build.log
```

**Review by rdlrt — APPROVED** [review]

**Pull request merged**